### DISCUSSION TITLE
How can D2 be more useful for academic use?

### DISCUSSION THREAD

**Discussion by alixander** [terrastruct/d2#244] [Q&A]
Now that D2 will support Latex in the next release (https://github.com/terrastruct/d2/pull/229), would love to hear from students, researchers, teachers, is there anything else that D2 can do to be more useful for academic use cases?

**Comment by Tom-stack3** [comment]
Hi,
I've been trying to use _D2_ to create some diagrams for academic presentations, but I can't seem to find a way to make the diagrams fit.
I get a wonderful diagram with _dagre_ but it's just too big horizontally to fit in a slide.
I guess that maybe if there was a way to control certain parts that affect the size of the diagram, like the padding inside boxes, or the length of arrows, it would help.

btw, _D2_ looks fantastic :)

**Reply by pleshevskiy** [reply]
I join in on this. Much needed)

**Reply by alixander** [reply]
i don't think padding would make much of a difference, but perhaps length of arrows we can toggle -- still, it will continue to grow in one direction as dagre is a hierarchical layout engine. 

i can imagine our wrapper around dagre doing something like splitting up multiple containers, run layout on each of them in individual directions, and then stitch back together. however there'd have to be limitations, like these containers can't have children connected to each other, which would suck. 

will see if arrow sizing control helps first

**Reply by Tom-stack3** [reply]
I saw that the default height of the boxes is `126`, and I thought that it would be helpful if there was an option to change that default height in the file.

For example, the following certainly won't fit on a slide, but with smaller boxes, it would fit wonderfully.

<img src="https://user-images.githubusercontent.com/76645845/206542253-b56bbc2a-ae46-4fef-8eb9-ae5f252e262f.svg" width="300">

So maybe that could help :)

**Comment by OpenFoam-User** [comment]
Hi @alixander 
D2 is great for diagrams, but it could be much more useful for Academic use it could provide Tikz-like power or Penrose, see for example, the gallery of Penrose: https://penrose.cs.cmu.edu/examples

Thank you

**Reply by alixander** [reply]
Thanks for the reply. These tools are at different levels of abstraction. The fundamental difference is that their syntax is telling you how to draw it (segment from this coordinate to this coordinate), whereas D2's syntax is describing the model. The types of diagrams one makes with each tool is almost entirely mutually exclusive. It's out of scope for this tool/language.
